pkg/config/remote_config.go

```diff
@@ -0,0 +1,51 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"cloud.google.com/go/firestore"
+)
+
+type RemoteConfig interface {
+	GetRemoteConfigTo(ctx context.Context, data interface{}) error
```

Comment:
A few notes:

1. This needs documentation. We specifically need to note that the result is read into the value of data, and that data must be a pointer. We could actually enforce it being a pointer with generics, although it makes the function less readable:

    ```go
    GetRemoteConfigTo[T any](ctx context.Context, target *T) error
    ```

2. Doesn't this also need the config path from which to load the config?

3. Nit: I'm not a huge fan of this name. "Get" and "Set" prefixes are generally discouraged: https://go.dev/doc/effective_go#Getters. I suggest we name this `LoadRemoteConfig` instead.

Reply:
Would `Unmarshal` be a better name? This is essentially unmarshal the whole firestore doc into a struct.

Reply:
@raserva, I followed the idea of firestore "DataTo" when I name "GetRemoteConfigTo" (fyi: https://pkg.go.dev/cloud.google.com/go/firestore#DocumentSnapshot.DataTo)

Reply:
>Doesn't this also need the config path from which to load the config?

ConfigPath is provided when initializing the RemoteConfig 
```
type FirestoreRemoteConfig struct {
	client      *firestore.Client
	docFullPath string
}
```

Reply:
》 Would Unmarshal be a better name? This is essentially unmarshal the whole firestore doc into a struct.
Is it a little bit general? I still want to emphasize this is for `RemoteConfig` here.

Reply:
How about `Load/Update`?

Reply:
For the `Get`, I can't figure out an elegant way to do unmarshal when we get one specific field

Reply:
> I'm a little worried about Get/Set in an interface. There's a high probability that would conflict with another interface we might want to implement.

I'm not following. How come it's a conflict if _another_ interface also have Get/Set func? Unless we want to embed them together.
The following way feels pretty clear to me:

```go
jvskeyCfg, err := NewFirestoreConfig(ctx, "my-project-id", "jvs/keyconfig")

// Unmarshal the whole config
var keyCfg KeyConfig
err := jvskeyCfg.Unmarshal(ctx, &keyCfg)

// Get a field by key
bar, err := jvskeyCfg.Get(ctx, "abc.foo")

// Set a field by key
err := jvskeyCfg.Set(ctx, "abc.foo", "bar")
```

Reply:
It's a conflict if another interface as a Get/Set with a different function signature. Then it's impossible to satisfy both interfaces.

Reply:
Interesting. I'd say in case we run into such a situation, we look harder for alternative names in the **other** interfaces so we don't get conflicts :)
Meanwhile I'm open for alternatives. I'm not a fan of `GetByKey`/`SetByKey` mainly because `key` is already a parameter which makes it redundant information.

Reply:
Viper use `Unmarshal/Get/Set` (https://github.com/spf13/viper). I just change the name as what Viper does.